### DISCUSSION TITLE
Experimenting with the package

### DISCUSSION THREAD

**Discussion by 4722794** [david96182/ninjemail#2] [General]
Hi,

I don't see much discussion here; so I'm just going to log my experience using this library.

I see a lot of promise, but the documentation is a little hard to go through. So far, I've not been able to make anything work.

Here's what I've done so far:

1. Created a captcha account
2. Created accounts for sms services
3. Ran a simple code to create a yahoo account

I'm constantly getting a TimeoutException.

Not sure how to tackle it

**Comment by 4722794** [comment]
Has this worked for anyone?

So far, I've had no success with any of the email.

Mostly I get timeout errors, with gmail it's "Reached error page"

**Comment by 4722794** [comment]
Typical error with yahoo creation
```python
{
	"name": "TimeoutException",
	"message": "Message: 
Stacktrace:
RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5
NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5
dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16
",
	"stack": "---------------------------------------------------------------------------
TimeoutException                          Traceback (most recent call last)
File ~/Desktop/workshop/ninjemail/ninjemail/email_providers/yahoo.py:141, in create_account(captcha_key, driver, sms_key, username, password, first_name, last_name, month, day, year, myyahoo)
    139 try:
    140     # funcaptcha challenge
--> 141     WebDriverWait(driver, WAIT).until(EC.frame_to_be_available_and_switch_to_it((By.ID, \"arkose-iframe\")))
    142     while True:

File ~/Desktop/workshop/ninjemail/venv/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:105, in WebDriverWait.until(self, method, message)
    104         break
--> 105 raise TimeoutException(message, screen, stacktrace)

TimeoutException: Message: 
Stacktrace:
RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5
NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5
dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16


During handling of the above exception, another exception occurred:

TimeoutException                          Traceback (most recent call last)
Cell In[2], line 1
----> 1 email, password = ninja.create_yahoo_account(use_proxy=False)

File ~/Desktop/workshop/ninjemail/ninjemail/ninjemail_manager.py:276, in Ninjemail.create_yahoo_account(self, username, password, first_name, last_name, birthdate, myyahoo, use_proxy)
    272 username, password, first_name, last_name, \\
    273     _, birthdate = generate_missing_info(username, password, first_name, last_name, '', birthdate)
    274 month, day, year = get_birthdate(birthdate)
--> 276 return yahoo.create_account(captcha_key,
    277                             driver, 
    278                             sms_key,
    279                             username, 
    280                             password, 
    281                             first_name, 
    282                             last_name,
    283                             month,
    284                             day,
    285                             year,
    286                             myyahoo)

File ~/Desktop/workshop/ninjemail/ninjemail/email_providers/yahoo.py:156, in create_account(captcha_key, driver, sms_key, username, password, first_name, last_name, month, day, year, myyahoo)
    153         logging.error(\"Captcha was not solved.\")
    154 except:
    155     # recaptcha challenge
--> 156     WebDriverWait(driver, WAIT).until(EC.frame_to_be_available_and_switch_to_it((By.ID, \"recaptcha-iframe\")))
    157     time.sleep(10)
    158     try:

File ~/Desktop/workshop/ninjemail/venv/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:105, in WebDriverWait.until(self, method, message)
    103     if time.monotonic() > end_time:
    104         break
--> 105 raise TimeoutException(message, screen, stacktrace)

TimeoutException: Message: 
Stacktrace:
RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5
NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5
dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16
"
}
```

**Comment by david96182** [comment]
Hi,

Thank you for testing my library and sharing your feedback. I appreciate your effort in experimenting with it. 

Regarding the error you're experiencing, it's important to note that the account for the captcha and SMS services must have a balance in order to function properly. Please ensure that you have sufficient balance in those accounts.

To achieve better results, it is strongly recommended to use your own proxies with the library. By using proxies, you can improve the performance and avoid potential connection issues. Additionally, the speed of your internet connection can also impact the library's performance. If you have a poor connection, it may result in slower page loading times and potential exceptions.

Furthermore, please keep in mind that this library is relatively new, having been created a few months ago. As a result, it may not cover all possible actions for every email provider, especially considering the variations based on country IP. Addressing these complexities can be challenging and time-consuming. That's why I encourage you to star and share the project, as it will attract more people who can collaborate and contribute, making the library more robust and comprehensive.

**Comment by 4722794** [comment]
Hello David;
Firstly, great idea. Starred and shared.

About your suggestions, I've put balance in both sms and captcha services. But I haven't yet been able to make it work.

Could you make a short video tutorial on generating a gmail account? 

This can also be helpful to the wider community when they start working with your library.

I'm happy to make the video, but I haven't been able to run it even once.

Let me know if I can be of help.

**Comment by 4722794** [comment]
Update:
Alright, this library definitely works.
Here were some of the quirks that helped me make it work:
1. Use chrome (better `undetected-chrome`) instead of firefox
2. Use a good proxy provider (smartproxy worked best)
3. smspool is a reliable and easy-to-use sms service (but very expensive) 
4. For some reason, Ninjemail has not yet worked in `headless` mode for me. It fails at the `input number` step*

Great work by the creator! @david96182 thanks for putting this together

* If the creator believes this could really be an issue - I will open up an issue else @david96182 could you please respond with best practices on using with headless mode?

**Comment by david96182** [comment]
Thanks!!!

> * If the creator believes this could really be an issue - I will open up an issue else @david96182 could you please respond with best practices on using with headless mode?

If more users experience the same error, then you can definitely open an issue. I have not received any other reports about it. The web drivers are configured to open in headless mode by default, so I'm not sure what could be the problem. It could be the version of the browser installed or the operating system. Also, please make sure you are not modifying any settings of the web drivers and provide more details.

We have a community on Telegram if you would like to join and discuss more about issues and the project in general. You can find the Telegram community [here](https://github.com/david96182/ninjemail?tab=readme-ov-file#community).

**Comment by david96182** [comment]
Do you still have problems with the headless mode?

**Comment by 4722794** [comment]
I'm just about to start testing; have joined your telegram group and will continue reporting there.

**Reply by VortexDevTech** [reply]
Hey does the repo still works?
